src/mapping/device/GinkgoRBFKernels.cpp

```diff
@@ -1,132 +1,188 @@
-                                                                           double *, double *, double *, precice::mapping::Gaussian, const RadialBasisParameters,
-                                                                           const std::size_t, const std::size_t, const bool, const unsigned int);
+namespace precice::mapping {
 
-template void create_rbf_system_matrix<double, precice::mapping::CompactThinPlateSplinesC2>(std::shared_ptr<const DefaultExecutor>, const std::size_t, const std::size_t, const std::size_t, const std::array<bool, 3>,
-                                                                                            double *, double *, double *, precice::mapping::CompactThinPlateSplinesC2, const RadialBasisParameters,
-                                                                                            const std::size_t, const std::size_t, const bool, const unsigned int);
+std::shared_ptr<gko::Executor> create_device_executor(const std::string &execName, bool enableUnifiedMemory)
+{
 
-template void create_rbf_system_matrix<double, precice::mapping::CompactPolynomialC0>(std::shared_ptr<const DefaultExecutor>, const std::size_t, const std::size_t, const std::size_t, const std::array<bool, 3>,
-                                                                                      double *, double *, double *, precice::mapping::CompactPolynomialC0, const RadialBasisParameters,
-                                                                                      const std::size_t, const std::size_t, const bool, const unsigned int);
+#ifdef KOKKOS_ENABLE_SERIAL
+  if (execName == "reference-executor") {
+    return gko::ext::kokkos::create_executor(Kokkos::Serial{});
+  }
+#endif
+  if (enableUnifiedMemory) {
+    return gko::ext::kokkos::create_executor(Kokkos::DefaultExecutionSpace{},
+                                             Kokkos::SharedSpace{});
+  } else {
+    return gko::ext::kokkos::create_executor(Kokkos::DefaultExecutionSpace{});
+  }
+}
 
-template void create_rbf_system_matrix<double, precice::mapping::CompactPolynomialC2>(std::shared_ptr<const DefaultExecutor>, const std::size_t, const std::size_t, const std::size_t, const std::array<bool, 3>,
-                                                                                      double *, double *, double *, precice::mapping::CompactPolynomialC2, const RadialBasisParameters,
-                                                                                      const std::size_t, const std::size_t, const bool, const unsigned int);
+namespace kernel {
+
+template <typename MemorySpace, typename EvalFunctionType>
+void create_rbf_system_matrix_impl(std::shared_ptr<const gko::Executor>      exec,
+                                   gko::ptr_param<GinkgoMatrix>              mtx,
+                                   const std::array<bool, 3>                 activeAxis,
```

Comment:
If you ever need more performance, turning this into a template argument could help.

Reply:
The `std::array`?

Reply:
The xyz bools, yes.

Reply:
What makes you assume so? The array has already a compile-time constant size, I don't see templates helping here with performance.

Reply:
I mean to turn the three bools into template arguments. So you end up with one function per valid combination, eliminating the inner loop.

Reply:
ah I see, noted!